fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -2126,13 +2139,26 @@ private static void parseInclusionParameter(Class<?> resourceType, FHIRSearchCon
         String resourceTypeAndParameterName;
         String searchParameterTargetType;
 
-        SearchParameter searchParm;
-        InclusionParameter newInclusionParm;
-        List<InclusionParameter> newInclusionParms = null;
-
         List<String> allowedIncludes = getSearchPropertyRestrictions(resourceType.getSimpleName(), SEARCH_PROPERTY_TYPE_INCLUDE);
         List<String> allowedRevIncludes = getSearchPropertyRestrictions(resourceType.getSimpleName(), SEARCH_PROPERTY_TYPE_REVINCLUDE);
 
+        // Parse inclusionKeyword into parameter name and modifier (if present).
+        Modifier modifier = null;
+        int index = inclusionKeyword.indexOf(SearchConstants.COLON_DELIMITER_STR);
```

Comment:
Perhaps move this logic into a method, similar to `boolean SearchUtil.isAllowed(...)` for consistency.

Reply:
I considered that, but this one's a little weird in that the modifier applies only to the _include and _revinclude search parameters rather than to a type of search parameter. Given that, I decided to just keep the logic inline in this method.